n4js-libs/packages/n4js-cli/src/n4js/n4jsc.n4js

```diff
@@ -27,10 +27,11 @@ import {getJavaCommand} from "JreProvider";
  */
 export public async function runN4jscSync(log: {function(string)} = npmlog.info) {
     const args = getArgsFromProcess();
-
-    let exitCode : int = await runN4jsc(args, log);
-    if (exitCode !== 0) {
+    try {
+        let exitCode : int = await runN4jsc(args, log);
         process.exit(exitCode);
+    } catch (error) {
+        process.exit(error as int);
```

Comment:
In case of unexpected exceptions, `error` might not be of type `int`. Maybe add a `typeof number` check?